src/util/locktime.rs

```diff
@@ -0,0 +1,516 @@
+// Rust Bitcoin Library
+// Written in 2022 by
+//     Tobin C. Harding <me@tobin.cc>
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! Provides type `LockTime` that implements the logic around nLockTime/OP_CHECKLOCKTIMEVERIFY.
+//!
+
+use core::{mem, fmt};
+use core::cmp::{PartialOrd, Ordering};
+use core::convert::TryFrom;
+
+use crate::consensus::encode::{self, Decodable, Encodable};
+use crate::io;
+
+/// The Threshold for deciding whether a lock time value is interpreted as height or time (see [Bitcoin Core]).
+///
+/// [Bitcoin Core]: https://github.com/bitcoin/bitcoin/blob/9ccaee1d5e2e4b79b0a7c29aadb41b97e4741332/src/script/script.h#L39
+pub const LOCKTIME_THRESHOLD: u32 = 500_000_000;
+
+/// A transaction with nLocktime==0 is able to be included immediately in any block.
+pub const ZERO: LockTime = LockTime::Blocks(Height(0));
+
+/// Returns true if `n` is a block height i.e., is less than 500,000,000.
+///
+/// Bitcoin is able to safely use this threshold because a block height greater than 500,000,000
+/// won't happen for approximately 9500 years.
+pub fn is_block_height(n: u32) -> bool {
+    n < LOCKTIME_THRESHOLD
+}
+
+/// Returns true if `n` is a UNIX timestamp i.e., is greater than or equal to 500,000,000.
+///
+/// Bitcoin is able to safely use this threshold because values of `n` under 500,000,000 represent
+/// times before 1986 which are, for obvious reasons, not useful within the Bitcoin network.
+pub fn is_block_time(n: u32) -> bool {
```

Comment:
Do we want these two `pub`? isn't it better to just convert to `LockTime` and handle it as enum?